integration/update_test.go

```diff
@@ -770,6 +770,66 @@ func TestUpdateMany(t *testing.T) {
 	}
 }
 
+func TestMultiFlag(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		filter bson.D
+		update bson.D
+		multi  string
+		stat   bson.D
+	}{
+		"MultiFalse": {
+			filter: bson.D{{"foo", "x"}},
+			update: bson.D{{"$set", bson.D{{"foo", "y"}}}},
+			multi:  "false",
+			stat:   bson.D{{"n", int32(1)}, {"nModified", int32(1)}, {"ok", float64(1)}},
+		},
+		"MultiTrue": {
+			filter: bson.D{{"foo", "x"}},
+			update: bson.D{{"$set", bson.D{{"foo", "y"}}}},
+			multi:  "true",
+			stat:   bson.D{{"n", int32(2)}, {"nModified", int32(2)}, {"ok", float64(1)}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx, collection := setup(t)
+
+			_, err := collection.InsertMany(ctx, []any{
+				bson.D{{"_id", "first"}, {"foo", "x"}},
+				bson.D{{"_id", "second"}, {"foo", "x"}},
+			})
+			require.NoError(t, err)
+
+			var actual bson.D
+
+			if tc.multi == "true" {
```

Comment:
Running `update` with `multi:true` via `RunCommand` was actually a good idea, otherwise I'm not sure what are we even testing here
@AlekSi WDYT?

Reply:
UpdateMany and UpdateOne are Go driver's methods that send `update` with `multi:true` and `multi:false` under the hood